### PR TITLE
Add ACR and AMR attributes to access token

### PR DESCRIPTION
**Pull request by mfjarvis**
## Summary
* Routine Change
* :exclamation: Breaking Change
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature
* :warning: Potential issues that might be caused by this change

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: proxies/live/apiproxy/policies/OAuthV2.GenerateAccessToken.xml [hunk 1]
```diff
@@ -13,6 +13,8 @@
       <Attribute name="id_token" ref="externalIdToken" display="true"/>
       <Attribute name="id_token-subject" ref="jwt.VerifyJWT.FromExternalIdToken.claim.subject" display="true"/>
       <Attribute name="id_token-issuer" ref="jwt.VerifyJWT.FromExternalIdToken.claim.issuer" display="true"/>
+      <Attribute name="id_token-acr" ref="jwt.VerifyJWT.FromExternalIdToken.claim.acr" display="true"/>
```
Comment (mfjarvis) [inline proxies/live/apiproxy/policies/OAuthV2.GenerateAccessToken.xml:16]: Are these important enough that we should expose them as additional attributes? We could just make people decode the id token if they need to access them

Comment (mfjarvis) [reply proxies/live/apiproxy/policies/OAuthV2.GenerateAccessToken.xml:16]: I think the question I'm really asking is: Will these only ever be used in the Signing Service API?

Comment (harryhmc) [reply proxies/live/apiproxy/policies/OAuthV2.GenerateAccessToken.xml:16]: how exactly are these being used in the signing service? i'm in favour of having the consumer have to decode the token if these claims aren't widely / frequently used by other APIs

File: proxies/live/apiproxy/policies/GenerateJWT.SimulatedAuth.xml [hunk 2]
```diff
@@ -8,4 +8,8 @@
     <Id>1234567</Id>
     <Value ref="private.dummy_secret_key"/>
   </SecretKey>
+  <AdditionalClaims>
+    <Claim name='acr'>AAL3_ANY</Claim>
+    <Claim name='amr' array='true'>N3_SMARTCARD</Claim>
```
[the remark this answers is not in the document]
Comment (mfjarvis) [reply proxies/live/apiproxy/policies/GenerateJWT.SimulatedAuth.xml:13]: No, the `array='true'` covers that

File: proxies/live/apiproxy/policies/OAuthV2.GenerateAccessToken.xml (hunk 1 above)
Comment (mfjarvis) [reply proxies/live/apiproxy/policies/OAuthV2.GenerateAccessToken.xml:16]: The answer at the moment is they're not being used at all.
Agreed. I think we need Rich to weigh in on this.

File: proxies/live/apiproxy/policies/DecodeJWT.FromGeneratedIdTokenSimulatedAuth.xml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<DecodeJWT async="false" continueOnError="false" enabled="true" name="DecodeJWT.FromGeneratedIdTokenSimulatedAuth">
```
[the remark this answers is not in the document]
Comment (mfjarvis) [reply proxies/live/apiproxy/policies/DecodeJWT.FromGeneratedIdTokenSimulatedAuth.xml:1]: Hadn't even considered it tbh. I'll take a look

[the remark this answers is not in the document]
Comment (mfjarvis) [reply proxies/live/apiproxy/policies/DecodeJWT.FromGeneratedIdTokenSimulatedAuth.xml:1]: Oh yeah, defaults to false anyway. I'll get rid of it

File: proxies/live/apiproxy/policies/GenerateJWT.SimulatedAuth.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (harryhmc) [reply proxies/live/apiproxy/policies/GenerateJWT.SimulatedAuth.xml:13]: cool, so if there were more entries in the array, how would that be represented here?

File: proxies/live/apiproxy/policies/OAuthV2.GenerateAccessToken.xml (hunk 1 above)
Comment (mfjarvis) [reply proxies/live/apiproxy/policies/OAuthV2.GenerateAccessToken.xml:16]: At least two APIs will use these attributes (EPS & Signing Service), and Rich suspects more will use them in future, so maybe worth exposing them?

File: proxies/live/apiproxy/policies/GenerateJWT.SimulatedAuth.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (mfjarvis) [reply proxies/live/apiproxy/policies/GenerateJWT.SimulatedAuth.xml:13]: I have no idea :joy:

[the remark this answers is not in the document]
Comment (mfjarvis) [reply proxies/live/apiproxy/policies/GenerateJWT.SimulatedAuth.xml:13]: It took me ages just to work out that this is how you're supposed to return an array

File: proxies/live/apiproxy/policies/OAuthV2.GenerateAccessToken.xml (hunk 1 above)
Comment (harryhmc) [reply proxies/live/apiproxy/policies/OAuthV2.GenerateAccessToken.xml:16]: sure, sounds like they wont be dead weight for the most part :+1:

**Review by harryhmc — APPROVED** [review]

**Review by alexhawdon — APPROVED** [review]

**Pull request merged**